src/Hasher.php

```diff
@@ -0,0 +1,53 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Utils;
+
+use EoneoPay\Utils\Exceptions\HashingFailedException;
+use EoneoPay\Utils\Interfaces\HasherInterface;
+
+class Hasher implements HasherInterface
+{
+    /**
+     * @param string $string
+     * @param null|integer $cost
+     *
+     * @return string
+     *
+     * @throws \EoneoPay\Utils\Exceptions\HashingFailedException
+     */
+    public function hash(string $string, ?int $cost = null): string
+    {
+        $options = [];
+
+        // Only set cost if specified, otherwise algorithm default is used
+        if ($cost) {
+            $options['cost'] = $cost;
+        }
+
+        $hash = \password_hash($string, self::ALGO, $options);
+
+        // False is mainly associated with *nix/system crypt library faults or poor configuration
+        if ($hash === false) {
```

Comment:
We use a return early methodology, so this should be negated and the exception should be thrown below:
```
if ($hash !== false) {
    return $hash;
}

throw new ...